Velaptor/Graphics/RenderMediator.cs

```diff
@@ -4,41 +4,224 @@
+    /// <param name="fontItemComparer">Compares 2 font batch items for the purpose of sorting.</param>
+    /// <param name="rectItemComparer">Compares 2 rect batch items for the purpose of sorting.</param>
+    /// <param name="lineItemComparer">Compares 2 line batch items for the purpose of sorting.</param>
+    public RenderMediator(
+        IReactableFactory reactableFactory,
+        IComparer<RenderItem<TextureBatchItem>> textureItemComparer,
+        IComparer<RenderItem<FontGlyphBatchItem>> fontItemComparer,
+        IComparer<RenderItem<RectBatchItem>> rectItemComparer,
+        IComparer<RenderItem<LineBatchItem>> lineItemComparer)
     {
         EnsureThat.ParamIsNotNull(reactableFactory);
+        EnsureThat.ParamIsNotNull(textureItemComparer);
+        EnsureThat.ParamIsNotNull(fontItemComparer);
+        EnsureThat.ParamIsNotNull(rectItemComparer);
+        EnsureThat.ParamIsNotNull(lineItemComparer);
 
-        this.pushReactable = reactableFactory.CreateNoDataReactable();
-
-        var batchEndName = this.GetExecutionMemberName(nameof(NotificationIds.RenderBatchEndedId));
+        this.pushReactable = reactableFactory.CreateNoDataPushReactable();
 
-        this.pushReactable.Subscribe(new ReceiveReactor(
-            eventId: NotificationIds.RenderBatchEndedId,
+        var batchEndName = this.GetExecutionMemberName(nameof(PushNotifications.BatchHasEndedId));
+        this.endBatchUnsubscriber = this.pushReactable.Subscribe(new ReceiveReactor(
+            eventId: PushNotifications.BatchHasEndedId,
             name: batchEndName,
             onReceive: CoordinateRenders));
+
+        var shutDownName = this.GetExecutionMemberName(nameof(PushNotifications.SystemShuttingDownId));
+        this.shutDownUnsubscriber = this.pushReactable.Subscribe(new ReceiveReactor(
+            eventId: PushNotifications.SystemShuttingDownId,
+            name: shutDownName,
+            onReceive: ShutDown));
+
+        this.texturePullReactable = reactableFactory.CreateTexturePullBatchReactable();
+        this.fontPullReactable = reactableFactory.CreateFontPullBatchReactable();
+        this.rectPullReactable = reactableFactory.CreateRectPullBatchReactable();
+        this.linePullReactable = reactableFactory.CreateLinePullBatchReactable();
+
+        this.textureRenderBatchReactable = reactableFactory.CreateRenderTextureReactable();
+        this.fontRenderBatchReactable = reactableFactory.CreateRenderFontReactable();
+        this.rectRenderBatchReactable = reactableFactory.CreateRenderRectReactable();
+        this.lineRenderBatchReactable = reactableFactory.CreateRenderLineReactable();
+
+        this.textureItemComparer = textureItemComparer;
+        this.fontItemComparer = fontItemComparer;
+        this.rectItemComparer = rectItemComparer;
+        this.lineItemComparer = lineItemComparer;
+
+        // Set all of the layers to a default value of the max value of int
```

Comment:
Should this be Set or Sets? 
```suggestion
        // Sets all of the layers to a default value of the max value of int
```